cinema/tests/test_movie_api.py

```diff
@@ -1,127 +1,126 @@
-        self.assertEqual(superman_movies.count(), 0)
-
-    def test_get_movie(self):
-        response = self.client.get("/api/cinema/movies/1/")
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(response.data["title"], "Titanic")
-        self.assertEqual(response.data["description"], "Titanic description")
-        self.assertEqual(response.data["duration"], 123)
-        self.assertEqual(response.data["genres"][0]["name"], "Drama")
-        self.assertEqual(response.data["genres"][1]["name"], "Comedy")
-        self.assertEqual(response.data["actors"][0]["first_name"], "Kate")
-        self.assertEqual(response.data["actors"][0]["last_name"], "Winslet")
-        self.assertEqual(
-            response.data["actors"][0]["full_name"], "Kate Winslet"
-        )
-
-    def test_get_invalid_movie(self):
-        response = self.client.get("/api/cinema/movies/100/")
-        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
-
-    def test_put_movie(self):
-        self.client.put(
-            "/api/cinema/movies/1/",
-            {
-                "title": "Watchman",
-                "description": "Watchman description",
-                "duration": 321,
-                "genres": [1, 2],
-                "actors": [1],
-            },
-        )
-        db_movie = Movie.objects.get(id=1)
-        self.assertEqual(
-            [db_movie.title, db_movie.description],
-            [
-                "Watchman",
-                "Watchman description",
-            ],
-        )
-        self.assertEqual(db_movie.title, "Watchman")
-
-    def test_delete_movie(self):
-        response = self.client.delete(
-            "/api/cinema/movies/1/",
-        )
-        db_movies_id_1 = Movie.objects.filter(id=1)
-        self.assertEqual(db_movies_id_1.count(), 0)
-        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
-
-    def test_delete_invalid_movie(self):
-        response = self.client.delete(
-            "/api/cinema/movies/1000/",
-        )
-        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from cinema.models import Actor, Genre, Movie
+
+
+class MovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        drama = Genre.objects.create(
+            name="Drama",
+        )
+        comedy = Genre.objects.create(
+            name="Comedy",
+        )
+        actress = Actor.objects.create(first_name="Kate", last_name="Winslet")
+        movie = Movie.objects.create(
+            title="Titanic",
+            description="Titanic description",
+            duration=123,
+        )
+        movie.genres.add(drama)
+        movie.genres.add(comedy)
+        movie.actors.add(actress)
+
+    def test_get_movies(self):
+        movies = self.client.get("/api/cinema/movies/")

```

Comment:
Consider using Django's `reverse` function to generate the URL for the API endpoint. This will make the test more robust against changes in the URL patterns.